DeviceWhoAmI.json

```diff
@@ -0,0 +1,41 @@
+{
+    "$id": "http://https://github.com/harp-tech/protocol/DeviceWhoAmI.json",
+    "$schema": "https://json-schema.org/draft-07/schema",
+    "definitions": {
+        "device": {
+            "properties": {
+                "contributor": {
+                    "type": "string"
+                },
+                "name": {
```

Comment:
JSON does not support using integers as map keys and it can be a bit clunky even for serializers. If we want to keep the schema YML-only this is fine, but otherwise we could have the device name as the key and replace `"name"` with `"whoAmI"` as a property of the device.

This is is also how we chose to do it for `registers` inside the device schema itself, so there is precedent:
https://github.com/harp-tech/reflex-generator/blob/main/schema/device.json

Reply:
I would be keen on keeping it as it enforces unique device id at the level of the yml file directly.

Reply:
Yup, this is what I thought you were going for. We could also do this check on a GitHub action that runs on the PR so we can be sure everything is unique before merging, but I am happy to go either way on this one so feel free to keep it as it is.